packages/ui/src/ui/store/actions/navigation/modals/cell-preview/dynamic-table.ts

```diff
@@ -0,0 +1,68 @@
+import type {CellPreviewActionType} from '../../../modals/cell-preview';
+import {
+    getCurrentRowKey,
+    getKeyColumns,
+    getYqlTypes,
+} from '../../../../selectors/navigation/content/table-ts';
+import Query from '../../../../../utils/navigation/content/table/query';
+import {YTApiId, ytApiV3Id} from '../../../../../rum/rum-wrap-api';
+import type {CancelTokenSource} from 'axios';
+
+export const getDynamicTableCellPath = ({
+    index,
+    path,
+    columnName,
+}: {
+    path: string;
+    columnName: string;
+    index: number;
+}): CellPreviewActionType => {
+    return (_, getState) => {
+        const yqlTypes = getYqlTypes(getState());
+        const key = getCurrentRowKey(getState(), index);
+        const keyColumns = getKeyColumns(getState());
+
+        const offset = Query.prepareKey(key, yqlTypes);
+
+        return Query.prepareQuery({
+            path,
+            offset,
+            offsetColumns: keyColumns,
+            columns: [columnName],
+            limit: 1,
+        });
+    };
+};
+
+export const getDynamicTableCliCommand = ({
```

Comment:
Are you sure it's action? there is no work with the API and there is no work with dispatch